### PR TITLE
Update sclo images configuration

### PR DESCRIPTION
**Pull request by hhorak**
There are few new images and dependencies. Also, in some cases (mariadb, mysql) the images need to use repository root directory as the context directory for `docker build` command.

### CODE REVIEW

**Comment by hhorak** [comment]
I guess we use a bit problematic layout of the repository, that's why the test fails. Reported as #220.

**Comment by kbsingh** [comment]
Can we skip the Mariadb ones for now, and get the rest in ? Then we can workout what changes are needed to the scl/mariadb repo and what we can support from the service side to unblock that.

**Comment by bamachrn** [comment]
@hhorak ^^ can we skip the Maridb ones for now?

**Comment by syamgk** [comment]
can we get the nodejs containers?
because am blocked on getting the latest nodejs images from registry.centos.org

**Comment by bamachrn** [comment]
@syamgk yes, we have got the change to service in repo. we should be able to get this in today. @mohammedzee1000 do we have anything else pending for this?

**Comment by mohammedzee1000** [comment]
@syamgk @hhorak @bamachrn Considering that the custom build context PR was merged in, the pipeline will be able to build fall the containers now. However there are a couple of actions that need to be done by sclo to enable this:

1. Make sure the cccp.yml files are in the git path, alongside the target file and not the root of the repository. For example. In case of s2i-core-centos7 (https://github.com/CentOS/container-index/pull/219/files#diff-98bb8d3e65f5c3c936b71ab5f675e105R4) and  s2i-base-centos7 (https://github.com/CentOS/container-index/pull/219/files#diff-98bb8d3e65f5c3c936b71ab5f675e105R172) the cccp.yml is sitting at the root of the repository (https://github.com/sclorg/s2i-base-container/blob/master/.cccp.yml).
2. Update this PR to add a parameter build-context which should contain a path, relative to the git-path.

For example: 
In case of the s2i-core-centos7, the index entry should look like this:

```
  - id: 1
    app-id: sclo
    job-id: s2i-core-centos7
    git-url: https://github.com/sclorg/s2i-base-container.git
    git-branch: master
    git-path: core/
    target-file: Dockerfile
    desired-tag: latest
    notify-email: hhorak@redhat.com
    build-context: ../
    depends-on:
      - centos/centos:latest
```

**Comment by hhorak** [comment]
Thanks for the changes and sorry for the delay. I'll try to make the suggested changes hopefully later today.

**Comment by mohammedzee1000** [comment]
FYI https://github.com/CentOS/container-pipeline-service/pull/365 this the PR which was merged recently for prebuild context

**Comment by hhorak** [comment]
Both issues, `job-id` in values and `build-context` should be fixed now.

File: index.d/sclo.yml [hunk 1]
```diff
@@ -1,443 +1,620 @@
...
   - id: 19
     app-id: sclo
-    job-id: python-33-centos7
-    git-url: https://github.com/sclorg/s2i-python-container
+    job-id: mariadb-100-centos7
+    git-url: https://github.com/sclorg/mariadb-container.git
     git-branch: master
-    git-path: 3.3/
-    target-file: Dockerfile
+    git-path: ./
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: ./ will not work here, considering dockerfile is in 10.0/

File: index.d/sclo.yml [hunk 2]
```diff
@@ -1,443 +1,620 @@
...
   - id: 22
     app-id: sclo
-    job-id: php-55-centos7
-    git-url: https://github.com/sclorg/s2i-php-container
+    job-id: mysql-56-centos7
+    git-url: https://github.com/sclorg/mysql-container.git
     git-branch: master
-    git-path: 5.5/
-    target-file: Dockerfile
+    git-path: ./
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: git-path should be 5.6/

File: index.d/sclo.yml [hunk 3]
```diff
@@ -1,443 +1,620 @@
...
   - id: 23
     app-id: sclo
-    job-id: php-56-centos7
-    git-url: https://github.com/sclorg/s2i-php-container
+    job-id: mysql-57-centos7
+    git-url: https://github.com/sclorg/mysql-container.git
     git-branch: master
-    git-path: 5.6/
-    target-file: Dockerfile
+    git-path: ./
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: git-path should be 5.7/

File: index.d/sclo.yml [hunk 4]
```diff
@@ -1,443 +1,620 @@
...
   - id: 20
     app-id: sclo
-    job-id: python-34-centos7
-    git-url: https://github.com/sclorg/s2i-python-container
+    job-id: mariadb-101-centos7
+    git-url: https://github.com/sclorg/mariadb-container.git
     git-branch: master
-    git-path: 3.4/
-    target-file: Dockerfile
+    git-path: ./
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: Same here git path should be 10.1/

File: index.d/sclo.yml [hunk 5]
```diff
@@ -1,443 +1,620 @@
...
   - id: 21
     app-id: sclo
-    job-id: python-35-centos7
-    git-url: https://github.com/sclorg/s2i-python-container
+    job-id: mariadb-102-centos7
+    git-url: https://github.com/sclorg/mariadb-container.git
     git-branch: master
-    git-path: 3.5/
-    target-file: Dockerfile
+    git-path: ./
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: Same git path should be 10.2/

File: index.d/sclo.yml [hunk 6]
```diff
@@ -1,443 +1,620 @@
...
     app-id: sclo
-    job-id: python-33-centos7
-    git-url: https://github.com/sclorg/s2i-python-container
+    job-id: mariadb-100-centos7
+    git-url: https://github.com/sclorg/mariadb-container.git
     git-branch: master
-    git-path: 3.3/
-    target-file: Dockerfile
+    git-path: ./
+    target-file: 10.0/Dockerfile
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: target-file should be Dockerfile

File: index.d/sclo.yml [hunk 7]
```diff
@@ -1,443 +1,620 @@
...
     app-id: sclo
-    job-id: python-34-centos7
-    git-url: https://github.com/sclorg/s2i-python-container
+    job-id: mariadb-101-centos7
+    git-url: https://github.com/sclorg/mariadb-container.git
     git-branch: master
-    git-path: 3.4/
-    target-file: Dockerfile
+    git-path: ./
+    target-file: 10.1/Dockerfile
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: target-file should be Dockerfile

File: index.d/sclo.yml [hunk 8]
```diff
@@ -1,443 +1,620 @@
...
     app-id: sclo
-    job-id: python-35-centos7
-    git-url: https://github.com/sclorg/s2i-python-container
+    job-id: mariadb-102-centos7
+    git-url: https://github.com/sclorg/mariadb-container.git
     git-branch: master
-    git-path: 3.5/
-    target-file: Dockerfile
+    git-path: ./
+    target-file: 10.2/Dockerfile
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: target-file should be Dockerfile

File: index.d/sclo.yml [hunk 9]
```diff
@@ -1,443 +1,620 @@
...
     app-id: sclo
-    job-id: php-55-centos7
-    git-url: https://github.com/sclorg/s2i-php-container
+    job-id: mysql-56-centos7
+    git-url: https://github.com/sclorg/mysql-container.git
     git-branch: master
-    git-path: 5.5/
-    target-file: Dockerfile
+    git-path: ./
+    target-file: 5.6/Dockerfile
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: target-file should be Dockerfile

File: index.d/sclo.yml [hunk 10]
```diff
@@ -1,443 +1,620 @@
...
     app-id: sclo
-    job-id: php-56-centos7
-    git-url: https://github.com/sclorg/s2i-php-container
+    job-id: mysql-57-centos7
+    git-url: https://github.com/sclorg/mysql-container.git
     git-branch: master
-    git-path: 5.6/
-    target-file: Dockerfile
+    git-path: ./
+    target-file: 5.7/Dockerfile
```
Comment (mohammedzee1000) [inline index.d/sclo.yml]: target-file should be Dockerfile

**Comment by hhorak** [comment]
@mohammedzee1000 Hope it's fixed now.

**Comment by mohammedzee1000** [comment]
LGTM

**Comment by bamachrn** [comment]
@hhorak @mohammedzee1000 I was just looking into the dockerfiles for the nodejs images. I found [this](https://github.com/sclorg/s2i-nodejs-container/blob/master/8/Dockerfile#L1) uses centos/s2i-base-centos7:latest as base. But here in [this PR](https://github.com/CentOS/container-index/pull/219/files#diff-98bb8d3e65f5c3c936b71ab5f675e105R4) we are building sclo/s2i-base-centos7:latest and sclo/s2i-core-centos7:latest. can we fix this ambiguity?
- may be we  use sclo/s2i-base-centos7:latest as FROM in the dockerfiles 
- or build  s2i-base-centos7:latest, s2i-core-centos7:latest under centos name space 
otherwise we are building the sclo/s2i-base-centos7:latest but it is not getting used and for building other image having the base pulled from dockerhub.

**Comment by mohammedzee1000** [comment]
@bamachrn thanks for pointing this out.

**Comment by hhorak** [comment]
Hm, this is tricky. All the sclo dockerfiles are currently built as `centos/something`, and pushed to docker hub by another (our internal) infrastructure, and we won't be able to switch to cccp entirely until we're sure it works fine for our images. Changing the FROM clause would break our workflow, so it's not possible to do it in the same Dockerfile..

Using another dockerfile with only different FROM clause is not very practical, plus it looks like changing the image names is not what we can do in the end -- the image names are hard-coded in various places (in templates, users' scripts and configuration, OpenShift example apps, ...).

So, it boils down to the bigger question -- how could we ever switch to images built in cccp, instead of our internal infrastructure.

At this point it looks to me like we simply need to use the same names, including the namespace. And we might even consider pushing the docker images to docker hub, so users find them.

Anyway, is it possible to build the images with `centos/` namespaces in cccp?

**Comment by bamachrn** [comment]
> Anyway, is it possible to build the images with centos/ namespaces in cccp?

@hhorak yes, and I believe that is the best solution to go for. This will also make sure images in dockerhub is matching the image names with registry.centos.org.

**Comment by mohammedzee1000** [comment]
If you add them here, that would get them into centos namespace https://github.com/CentOS/container-index/blob/master/index.d/centos.yml . If you plan to add them there instead, I would recommend putting a comment to indicate that these are sclorg images

```
# Sclorg images 
Sclorg index entries here
# Sclorg images ends
```

**Comment by hhorak** [comment]
Should I request an approval from somebody before submitting the PR for the `centos.yml` or can I do it right away?

**Comment by bamachrn** [comment]
@hhorak please go ahead adding them in `centos.yml`

**Comment by hhorak** [comment]
Images now moved to `centos.yml`.

**Comment by bamachrn** [comment]
@hhorak we need to update the appid to `centos` from `sclo` for the entries added.

**Comment by hhorak** [comment]
Sure, sorry. Should be better now.

**Comment by mohammedzee1000** [comment]
@hhorak LGTM. I will merge this as soon as we are done deploying the latest master. :)

**Pull request merged**